### ISSUE TITLE
Algolia hierarchy doesn't seem to be working

### ISSUE THREAD

**Issue by lilaconlee** [cypress-io/cypress-documentation#987]
Looks like where hierarchy should be in search, it just repeats the page title even if the content in the search result isn't top level.

<img width="525" alt="screen shot 2018-09-26 at 12 09 19 pm" src="https://user-images.githubusercontent.com/7514489/46096530-10b66480-c185-11e8-9490-32dedbfe90bd.png">

Above, I'd expect the first result to be "Syntax" and the best practices hit to be "Selecting Elements".

**Comment by brian-mann** [comment]
@bahmutov pretty much worked on the entirety of this thing - ping him for updates to get sync'd with why it's doing this. It's been mentioned before but I remember there being something challenging about the parsing of it. Not sure.

**Comment by bahmutov** [comment]
Not sure what you expect here instead. Algolia is _really_ strongly preferring text matches in h1, h2, h3 headers to text matches in the body of the page. Thus we need to pick good headings, and in the screenshot, the top result is the one with the heading, right?

My personal impression of Algolia was that it was like a black box, and it was hard to predict what exactly would be returned, thus once we got it where things were found well enough, we did not want to touch it again.
